File: src/game/level/world2/World2Level3.java
package game.level.world2;

import game.entity.enemies.boss.Boss;
import game.entity.enemies.npcs.*;
import game.entity.models.Enemy;
import game.level.Level;
import game.level.world1.World1Level;

public class World2Level3 extends World2Level {
    @Override
    public int getLevelId() {
        return 3;
    }

    @Override
    public int startEnemiesCount() {
        return 13;
    }

    @Override
    public Class<? extends Enemy>[] availableEnemies() {
        return new Class[]{
                TankEnemy.class,
                FastEnemy.class,
        };
    }

    @Override
    public Class<? extends Level> getNextLevel() {
        return World2Level4.class;
    }
}

File: src/game/engine/level/levels/world2/World2Level3.kt
package game.engine.level.levels.world2

import game.engine.world.domain.entity.actors.impl.enemies.npcs.fast_enemy.FastEnemy
import game.engine.world.domain.entity.actors.impl.enemies.npcs.tank.TankEnemy
import game.engine.world.domain.entity.actors.abstracts.enemy.Enemy
import game.engine.level.levels.Level
import game.engine.level.levels.StoryLevel
import game.engine.level.info.model.LevelInfo
import game.engine.level.info.imp.World2levelInfo

class World2Level3 : StoryLevel() {
    override val info: LevelInfo
        get() = object: World2levelInfo(this) {
            override val levelId: Int get() = 3
            override val startEnemiesCount: Int get() = 13
            override val availableEnemies: Array<Class<out Enemy>> get() = arrayOf(TankEnemy::class.java, FastEnemy::class.java)
            override val nextLevel: Class<out Level?> get() = World2Level4::class.java
        }
}